server/internal/infrastructure/gcp/cloudtasks.go

```diff
@@ -5,74 +5,64 @@ import (
 	"encoding/json"
 
 	cloudtasks "cloud.google.com/go/cloudtasks/apiv2"
+	"github.com/googleapis/gax-go/v2"
 	"github.com/reearth/reearth-cms/server/internal/usecase/gateway"
-	"github.com/reearth/reearth-cms/server/pkg/id"
-	"github.com/reearth/reearth-cms/server/pkg/id/idx"
 	"github.com/reearth/reearth-cms/server/pkg/task"
-	"google.golang.org/api/option"
 	taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2"
 )
 
 type TaskRunner struct {
 	queuePath     string
 	subscriberURL string
-	credFilePath  string
+	c             *cloudtasks.Client
 }
 
-func NewTaskRunner(c *CloudTasksConfig, opts ...TaskRunnerOption) (gateway.TaskRunner, error) {
-	opts2 := defaultTaskRunnerOptions()
-	for _, o := range opts {
-		o.Apply(opts2)
+func NewTaskRunner(ctx context.Context, c *CloudTasksConfig) (gateway.TaskRunner, error) {
+	qURL, err := c.buildQueueUrl()
+	if err != nil {
+		return nil, err
 	}
 
-	qURL, err := c.buildQueueUrl()
+	cl, err := cloudtasks.NewClient(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	return &TaskRunner{
 		queuePath:     qURL,
+		c:             cl,
 		subscriberURL: c.SubscriberURL,
-		credFilePath:  opts2.credFilePath,
 	}, nil
 }
 
 // Run implements gateway.TaskRunner
-func (t *TaskRunner) Run(ctx context.Context, p task.Payload) (id.TaskID, error) {
-
-	client, closeFn, err := t.client(ctx)
-	if err != nil {
-		return idx.ID[id.Task]{}, err
-	}
-	defer closeFn()
-
+func (t *TaskRunner) Run(ctx context.Context, p task.Payload) error {
 	bPayload, err := json.Marshal(p.DecompressAsset.Payload())
 	if err != nil {
-		return idx.ID[id.Task]{}, err
+		return err
 	}
 	req := t.buildRequest(t.subscriberURL, bPayload)
 
-	_, err = client.CreateTask(ctx, req)
+	_, err = t.CreateTask(ctx, req)
 	if err != nil {
-		return idx.ID[id.Task]{}, err
+		return err
```

Comment:
```suggestion
		return rerror.InternalBy(err)
```

Errors describing technical details should be hidden from the user for security reasons. Wrapping with this function will output the error in the response and console in the development environment, but will hide the details of the error from the user in the production environment.

Reply:
OK! Thanks for letting me know that.